wagtail/documents/views/documents.py

```diff
@@ -176,10 +183,16 @@ def edit(request, document_id):
             # Reindex the document to make sure all tags are indexed
             search_index.insert_or_update_object(doc)
 
+            edit_url = reverse('wagtaildocs:edit', args=(doc.id,))
+            redirect_url = 'wagtaildocs:index'
+            if next_url:
+                edit_url += "?" + urlencode({"next": next_url})
+                redirect_url = next_url
```

Comment:
What's the advantage of redirecting via the edit view rather than redirecting directly?